spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -338,6 +338,9 @@ protected void doStop() {
 	protected void actualStart() throws Exception {
 		this.aborted = false;
 		this.hasStopped = false;
+		if (getPrefetchCount() < this.messagesPerAck) {
+			setPrefetchCount(this.messagesPerAck);
```

Comment:
How is it going to work if `prefetchCount > messagesPerAck`?
What will happen to those prefetched messages which are not handled because we have reached `messagesPerAck`?
Thanks